engine/console/game.go

```diff
@@ -143,12 +149,61 @@ var defColors = map[tcell.Color]struct{}{
 func (game *Game) handleGameEvent(event interface{}) {
 	switch event := event.(type) {
 	case core.PlayerStarts:
-		game.numAliveSnakes = len(event.Players)
+		game.NumAliveSnakes = len(event.Players)
+		game.Snakes = make(map[int]*Snake)
 		for id, start := range event.Players {
-			game.snakes = append(game.snakes, Snake{id: id, head: start, style: genSnakeStyle(defColors)})
+			game.Snakes[id] = &Snake{alive: true, Head: start, Style: genSnakeStyle(defColors)}
 		}
 	case core.NewFood:
-		game.food = append(game.food, event.Pos)
+		game.Food = append(game.Food, event.Pos)
+	case core.PlayerMove:
+		for id, dir := range event.Moves {
+			prevHead := game.Snakes[id].Head
+			// move snake's head
+			switch dir {
+			case core.Up:
+				game.Snakes[id].Head.Y--
+			case core.Left:
+				game.Snakes[id].Head.X--
+			case core.Right:
+				game.Snakes[id].Head.X++
+			case core.Down:
+				game.Snakes[id].Head.Y++
+			default:
+				panic("the value of direction is unknown")
+			}
+
+			// move snakes body
+			if len(game.Snakes[id].Body) == 0 {
+				continue
+			}
+			for i := len(game.Snakes[id].Body) - 1; i > 0; i-- {
+				game.Snakes[id].Body[i] = game.Snakes[id].Body[i - 1]
+			}
+			game.Snakes[id].Body[0] = prevHead
+		}
+	case core.FoodEaten:
+		idx := 0
+		for i, food := range game.Food {
+			if food.X == event.Pos.X && food.Y == event.Pos.Y {
+				idx = i
+				break
+			}
+		}
+		game.Food[idx] = game.Food[len(game.Food) - 1]
+		game.Food[len(game.Food) - 1] = core.Coord{}
```

Comment:
Should not be needed... But it seems that game.Food should be a map too... You can delete elements from a map using `delete(game.Food, idx)`.